testserver/testserver.go

```diff
@@ -408,30 +443,63 @@ func NewTestServer(opts ...TestServerOpt) (TestServer, error) {
 		storeArg = fmt.Sprintf("--store=type=mem,size=%.2f", serverArgs.storeMemSize)
 	}
 
-	args := []string{
+	nodes := make([]nodeInfo, serverArgs.numNodes)
+	var initArgs []string
+	for i := 0; i < serverArgs.numNodes; i++ {
+		nodes[i].state = stateNew
+		nodes[i].listeningURLFile = filepath.Join(baseDir, fmt.Sprintf("listen-url%d", i))
+		if serverArgs.numNodes > 1 {
+			nodes[i].startCmdArgs = []string{
+				serverArgs.cockroachBinary,
+				startCmd,
+				secureOpt,
+				storeArg + strconv.Itoa(i),
+				fmt.Sprintf("--listen-addr=localhost:%d", 26257+i),
+				fmt.Sprintf("--http-addr=localhost:%d", 8080+i),
+				"--listening-url-file=" + nodes[i].listeningURLFile,
+				fmt.Sprintf("--join=localhost:%d,localhost:%d,localhost:%d", 26257, 26258, 26259),
```

Comment:
A super minor nit: here for `--join` the params seems hard-coded. I wonder if we want it more general. (Though we may have to involve a tedious for loop)

Reply:
I'm going to leave this as hardcoded for now, the ports have to correspond to the `listen-addr` port which always starts at `26257` and ends at `26259` for now so hardcoding always works right now but if we allow the user to pick the `listen-addr` port then we'll have to update this.